server/mediaserver_test.go

```diff
@@ -389,24 +381,28 @@ func TestCreateRTMPStreamHandler(t *testing.T) {
 	expectedSid := core.MakeStreamIDFromString("ghijkl", "secretkey")
 	u, _ := url.Parse("rtmp://localhost/" + expectedSid.String()) // with key
 	if sid := createSid(u); sid.StreamID() != expectedSid.String() {
-		t.Error("Unexpected streamid")
+		t.Error("Unexpected streamid", sid.StreamID())
 	}
 	u, _ = url.Parse("rtmp://localhost/stream/" + expectedSid.String()) // with stream
 	if sid := createSid(u); sid.StreamID() != expectedSid.String() {
 		t.Error("Unexpected streamid")
 	}
 	expectedMid := "mnopq"
-	key := hex.EncodeToString(core.RandomIdGenerator(StreamKeyBytes))
+	rand.Seed(123)
```

Comment:
It's a bit unfortunate we can't override the RNG anymore, because the side effects from the RNG make thinking about the tests harder if we have to take care to reset the internal state. What do you think about reinstating the old monkey-patching behavior?

Reply:
I would really prefer not to keep functions in the variables if it is avoidable - it makes reasoning about code harder. Here it makes tests slightly harder, but I think it's not complicated enough to resort to monkey-patching.

Reply:
How does it make reasoning about the code harder, since any changes are local changes? And it does make the tests easier  - because they are local changes.

Reply:
Where it is local? It will be global variable containing function, which can be changed anywhere in the code.
Tests gets read much more rarely than main code.

Reply:
Overriding the RNG is local to the test function. The test function is also the only place we'd do that, which makes the test code cleaner. What is the concern with making reasoning more difficult? Do we have other options?

Reply:
When I look at `var RandomIdGenerator = func(length uint) []byte {` - I don't know where it is overwritten, I forced to search through all the code to find out that it is done only in test function.

Reply:
We can add a comment to the RandomIdGenerator definition to say that it should only be overwritten during testing.

Overriding the RNG was done for a reason. The original behavior matched your change - we'd reset the seed for each test invocation - but it made tests harder to maintain, and we encountered problems with the RNG being called in *other* parts of the code, leading to hard-to-trace test breakage. See this commit: https://github.com/livepeer/go-livepeer/commit/c50c7b7995594b36ee64295e40970981b8970a6a

Reply:
@j0sh OK, I've put `RandomIdGenerator` back to variable, you can check it again.